OctopusDSC/DSCResources/cOctopusServerSslCertificate/cOctopusServerSslCertificate.psm1

```diff
@@ -0,0 +1,145 @@
+# dot-source the helper file (cannot load as a module due to scope considerations)
+. (Join-Path -Path (Split-Path (Split-Path $PSScriptRoot -Parent) -Parent) -ChildPath 'OctopusDSCHelpers.ps1')
+
+# Create constant for Octopus Deploy server Application ID
+$octopusServerApplicationId = "{E2096A4C-2391-4BE1-9F17-E353F930E7F1}"
+
+function Get-CurrentSSLBinding {
+    param([string] $ApplicationId,
+    [string]$Port)
+
+    $certificateBindings = (& netsh http show sslcert) | select-object -skip 3 | out-string
+    $newLine = [System.Environment]::NewLine
+    $certificateBindings = $certificateBindings -split "$newLine$newLine"
+    $certificateBindingsList = foreach ($certificateBinding in $certificateBindings) {
+        if ($certificateBinding -ne "") {
+            $certificateBinding = $certificateBinding -replace "  ", "" -split ": "
+            [pscustomobject]@{
+                IPPort          = ($certificateBinding[1] -split "`n")[0]
+                CertificateHash = ($certificateBinding[2] -split "`n" -replace '[^a-zA-Z0-9]', '')[0]
+                AppID           = ($certificateBinding[3] -split "`n")[0]
+                CertStore       = ($certificateBinding[4] -split "`n")[0]
+            }
+        }
+    }
+
+    return ($certificateBindingsList | Where-Object {($_.AppID.Trim() -eq $ApplicationId) -and ($_.IPPort.Trim() -eq "{0}:{1}" -f "0.0.0.0", $Port) })
+}
+
+function Get-TargetResource {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSDSCUseVerboseMessageInDSCResource", "")]
+    param (
+        [Parameter(Mandatory)]
+        [string] $InstanceName,
+        [ValidateSet("Present", "Absent")]
+        [string] $Ensure = "Present",
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory)]
+        [string] $Thumbprint,
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory)]
+        [string] $StoreName,
+        [ValidateNotNullOrEmpty()]
+        [int] $Port = 443
+    )
+
+    $existingSSLConfig = (Get-CurrentSSLBinding -ApplicationId $octopusServerApplicationId -Port $Port -IPAddress $IPAddress)
+
+    if ($null -ne $existingSSLConfig) {
+        $existingSSLPort = [int](($existingSSLConfig.IPPort).Split(":")[1])
+        $existingSSLThumbprint = $existingSSLConfig.CertificateHash.Trim()
```

Comment:
Bit weird we are using "thumbprint" and "hash" hee - would be good to make these consistent